### PR TITLE
Tooltips; Colours

### LINKED ISSUE #118: Tooltips to explain what some of these cryptically named buttons do
Particularly "Push tags", "Pull tags", "Sync tags", etc.

### LINKED ISSUE #125: Modal colour should get pulled out into UIColour
Can be used on loader and #118

### PR DESCRIPTION
**Pull request by braxtonhall**
resolves #125 
toward #118 
Kapture 2022-11-04 at 19 29 55

### CODE REVIEW

File: extension/ts/main/ui/tooltip.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	colour?: UIColour;
+	position?: TooltipPosition;
+	text: string;
+}
+
+const tooltipped = <T extends HTMLElement>(
+	element: T,
+	{text, colour = UIColour.GREY, position = TooltipPosition.TOP}: TooltipOptions
+): T => {
+	// This MiiIIIght not age well. Probably smarter to add an invisible anchor point inside element
```
Comment (adirar111) [inline extension/ts/main/ui/tooltip.ts:22]: i h8 css -- if it breaks we can revisit?

File: extension/ts/main/extensions/util/finderExtension.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import {loaderOverlaid} from "../../ui/loadingIndicator";
 import {onFormRender} from "../../entities/bookForm";
 import {createButton} from "../../ui/button";
 import {Finder, FinderParameters} from "../../services/finder/finder";
+import {UIColour} from "../../ui/colour";
```
Comment (adirar111) [inline extension/ts/main/extensions/util/finderExtension.ts:6]: is this a real linter err

File: extension/ts/main/extensions/author/authorPage/pull.ts [hunk 3]
```diff
@@ -23,14 +24,14 @@ const uncertainTagModal =
 		createModal({
 			text: `Additional tag${strings.s}`,
 			subText: [...uncertainTags],
-			colour: ModalColour.BLUE,
+			colour: UIColour.BLUE,
```
Comment (adirar111) [inline extension/ts/main/extensions/author/authorPage/pull.ts:27]: mmmmmmmmmmmmmmmmmm

**Review by adirar111 — APPROVED** [review]
fire

File: extension/ts/main/ui/tooltip.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	RIGHT = "vbl-tool-tip-right",
+}
+
+interface TooltipOptions {
+	colour?: UIColour;
+	position?: TooltipPosition;
+	text: string;
+}
+
+const tooltipped = <T extends HTMLElement>(
```
Comment (adirar111) [inline extension/ts/main/ui/tooltip.ts:18]: `tooltipped` lmao

File: extension/ts/main/ui/tooltip.ts (hunk 1 above)
Comment (braxtonhall) [reply extension/ts/main/ui/tooltip.ts:22]: yeah i think we just would need another intermediate container

File: extension/ts/main/extensions/util/finderExtension.ts (hunk 2 above)
Comment (braxtonhall) [reply extension/ts/main/extensions/util/finderExtension.ts]: fixed

**Pull request merged**